src/schema/contentPreference.ts

```diff
@@ -0,0 +1,285 @@
+import { IResolvers } from '@graphql-tools/utils';
+import { traceResolvers } from './trace';
+import { AuthContext, BaseContext } from '../Context';
+import { ContentPreference } from '../entity/contentPreference/ContentPreference';
+import { toGQLEnum } from '../common';
+import {
+  ContentPreferenceStatus,
+  ContentPreferenceType,
+} from '../entity/contentPreference/types';
+import { followEntity, unfollowEntity } from '../common/contentPreference';
+import { GQLEmptyResponse, offsetPageGenerator } from './common';
+import graphorm from '../graphorm';
+import { Connection, ConnectionArguments } from 'graphql-relay';
+
+export type GQLContentPreference = Pick<
+  ContentPreference,
+  'referenceId' | 'userId' | 'type' | 'createdAt' | 'status'
+>;
+
+export const typeDefs = /* GraphQL */ `
+  ${toGQLEnum(ContentPreferenceType, 'ContentPreferenceType')}
+
+  ${toGQLEnum(ContentPreferenceStatus, 'ContentPreferenceStatus')}
+
+  enum FollowStatus {
+    follow
+    subscribed
+  }
+
+  type ContentPreference {
+    referenceId: ID!
+
+    user: User!
+
+    type: ContentPreferenceType!
+
+    createdAt: DateTime!
+
+    status: ContentPreferenceStatus!
+  }
+
+  type ContentPreferenceEdge {
+    node: ContentPreference!
+
+    """
+    Used in \`before\` and \`after\` args
+    """
+    cursor: String!
+  }
+
+  type ContentPreferenceConnection {
+    pageInfo: PageInfo!
+    edges: [ContentPreferenceEdge!]!
+  }
+
+  extend type Query {
+    """
+    Content preference status
+    """
+    contentPreferenceStatus(
+      """
+      Id of the entity
+      """
+      id: ID!
+      """
+      Entity type (user, source..)
+      """
+      entity: ContentPreferenceType!
+    ): ContentPreference @auth
+
+    """
+    Who follows user
+    """
+    userFollowers(
```

Comment:
This can be multiple entities right?
Just raising it because below is called:

- userFollowing
- this should maybe be `userFollows`
- And have description: `What follows user`

Or certain entities should not be allowed here if its only users.

Reply:
> Or certain entities should not be allowed here if its only users.

👍 agree with this

Reply:
Do we ever plan in our roadmap to show which users follows a specific source? just thinking as if so, this could be another abstraction where we don't specify user, rather entityFollowers(this would be any type of entity to users) and userFollowing can stay the same as it doesn't really make sense for any other entity to be following rather then just user.

Reply:
I am don't understand the naming issue, we have two queries userFollowers (followers of the user), userFollowing (entities that user is following)?

Reply:
All good on my side

Reply:
followers just sounds like it's always `users` 
only `userFollows` but that could again by confusing, so just leave it for now :)